libraries/AP_Compass/AP_Compass_Backend.cpp

```diff
@@ -90,13 +90,15 @@ void AP_Compass_Backend::correct_field(Vector3f &mag, uint8_t i)
 
 #if AP_COMPASS_DIAGONALS_ENABLED
     // apply eliptical correction
-    Matrix3f mat(
-        diagonals.x, offdiagonals.x, offdiagonals.y,
-        offdiagonals.x,    diagonals.y, offdiagonals.z,
-        offdiagonals.y, offdiagonals.z,    diagonals.z
-    );
-
-    mag = mat * mag;
+    if (!diagonals.is_zero()) {
```

Comment:
Any element being zero would be bad in here.  I wonder if it would be possible to get into that situation with loaded parameters.

Since this equivalent to what we had pre-regression, it really doesn't matter.